pkg/syncer/volume_health_reconciler.go

```diff
@@ -91,16 +148,74 @@ func NewVolumeHealthReconciler(
 		svcPVCSynced:        svcPVCInformer.Informer().HasSynced,
 		svcClaimQueue:       svcClaimQueue,
 		supervisorNamespace: supervisorNamespace,
+		volumeHandleToPVs: &volumeHandleToPVs{
+			RWMutex: &sync.RWMutex{},
+			items:   make(map[string][]string),
+		},
 	}
 
 	svcPVCInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
 		AddFunc:    rc.svcAddPVC,
 		UpdateFunc: rc.svcUpdatePVC,
 	}, resyncPeriod)
 
+	tkgPVInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
+		AddFunc:    nil,
+		UpdateFunc: rc.tkgUpdatePV,
+		DeleteFunc: rc.tkgDeletePV,
+	}, resyncPeriod)
+
+	// Initialize volumeHandleToPVs map with existing PV's
+	// Since this mapping is lost across pvCSI restarts, it needs to be
+	// initialized with existing values to prevent missing an SVC-PVC
+	// event due to absence of an entry in the map.
+	ctx, log := logger.GetNewContextWithLogger()
+
+	pvs, err := rc.tkgKubeClient.CoreV1().PersistentVolumes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		log.Fatalf("failed to list TKG PV's with error: %+v", err)
+	}
+	for _, pv := range pvs.Items {
```

Comment:
So, what happens to the TKG PVs that are created and bound after all the TKG PVs are listed and before the tkgPVInformer is started?